### DISCUSSION TITLE
Duplicate landscape images tracker

### DISCUSSION THREAD

**Discussion by Amazinite** [endless-sky/endless-sky#7107] [General]
### Problem Description

~~We currently have 332 separate landscape images, but 374 landable planets in the game. This means that some landscape images are used more than one time.~~ After #6450, the count is now 413 landscape images to 374 landable planets, removing almost all of the duplicate uses of landscape images. This issue will stay open to track the usage rate of each landscape image, mainly to show which landscape images still haven't been used.

### Related Issue Links

#3465 suggested the creation of images that better match planet descriptions, which would potentially free up some images to be used elsewhere.
#4511 tracks missing wiki info.

### Desired Solution

Remove as many duplicates as possible.
<details><summary>Table of landscape images along with how many times they're used, ordered by descending usage.</summary>

Image | Usage
---|---
badlands0 | 1
badlands1 | 1
badlands2 | 1
badlands3 | 1
badlands4 | 1
badlands5 | 1
badlands6 | 1
badlands7 | 1
badlands8 | 1
badlands9-harro | 1
badlands10 | 1
badlands11 | 1
badlands12 | 1
beach0 | 1
beach1 | 1
beach2 | 1
beach3 | 1
beach4 | 1
beach5 | 1
beach6 | 1
beach7-sfiera | 1
beach8-sfiera | 1
beach9-sfiera | 1
beach10-sfiera | 1
beach11-harro | 1
beach12-harro | 1
beach13 | 1
beach14 | 1
bwerner0 | 1
bwerner1 | 1
bwerner2 | 1
bwerner3 | 1
bwerner4 | 1
bwerner5 | 1
bwerner6 | 1
bwerner7 | 1
bwerner8 | 1
canyon0 | 1
canyon1 | 1
canyon01 | 1
canyon2 | 1
canyon02 | 1
canyon3 | 1
canyon03 | 1
canyon4 | 1
canyon04 | 1
canyon5 | 1
canyon05 | 1
canyon6 | 1
canyon7 | 1
canyon8 | 1
canyon9 | 1
canyon10-harro | 1
canyon11 | 1
canyon12 | 1
canyon13 | 1
canyon14 | 1
canyon15 | 1
city0 | 1
city1 | 1
city2 | 1
city3 | 1
city4 | 1
city5 | 1
city6 | 1
city7 | 1
city8 | 1
city9 | 1
city10 | 1
city11 | 1
city12 | 1
city13-sfiera | 1
city14-sfiera | 1
city18-iridium | 1
desert0 | 1
desert1 | 1
desert2 | 1
desert3 | 1
desert4 | 1
desert5 | 1
desert6 | 1
desert7 | 1
desert8-sfiera | 1
desert10-harro | 1
desert11 | 1
desert12 | 1
desert13 | 1
dmottl0 | 1
dmottl1 | 1
dmottl2 | 1
dmottl3 | 1
dmottl4 | 1
dmottl5 | 1
dune0 | 1
dune1 | 1
dune2 | 1
dune3 | 1
dune4 | 1
dune5 | 1
dune6-harro | 1
earthrise | 1
fields0 | 1
fields1 | 1
fields2 | 1
fields3 | 1
fields4 | 1
fields5 | 1
fields6 | 1
fields7 | 1
fields8 | 1
fields9 | 1
fields10-sfiera | 1
fields11-sfiera | 1
fields12-sfiera | 1
fields13 | 1
fields13-sfiera | 1
fields14 | 1
fields15 | 1
fog0 | 1
fog1 | 1
fog2 | 1
fog3 | 1
fog4 | 1
fog5 | 1
fog6 | 1
fog7 | 1
fog8 | 1
fog9 | 1
fog10 | 1
fog11 | 1
forest0 | 1
forest1 | 1
forest2 | 1
forest3 | 1
forest4 | 1
forest5 | 1
forest6 | 1
forest7 | 1
forest8 | 1
forest9 | 1
garden1 | 1
hills0 | 1
hills1 | 1
hills2 | 1
hills3 | 1
hills4 | 1
hills5-sfiera | 1
hills6-sfiera | 1
hills7-harro | 1
hills8 | 1
industrial0-iridium | 1
lava0 | 1
lava1 | 1
lava2 | 1
lava3 | 1
lava4 | 1
lava5 | 1
lava6 | 1
lava7-sfiera | 1
lava8 | 1
lava9 | 1
lava10 | 1
lava11 | 1
lava12 | 1
lava13 | 1
loc0 | 1
loc1 | 1
loc2 | 1
loc3 | 1
mars0 | 1
mars1 | 1
mars2 | 1
mars3 | 1
mfield0 | 1
mfield1 | 1
mfield2 | 1
mfield3 | 1
mfield4 | 1
mfield5 | 1
mountain0 | 1
mountain1 | 1
mountain2 | 1
mountain3 | 1
mountain4 | 1
mountain5 | 1
mountain6 | 1
mountain7 | 1
mountain8 | 1
mountain9 | 1
mountain10-sfiera | 1
mountain11-sfiera | 1
mountain12-sfiera | 1
mountain13-sfiera | 1
mountain15-sfiera | 1
mountain16-striker | 1
mountain17-harro | 1
mountain18-harro | 1
mountain19-harro | 1
mountain20-harro | 1
mountain21-harro | 1
mountain22-spfld | 1
mountain23-spfld | 1
mountain24-spfld | 1
mountain25-spfld | 1
mountain26 | 1
myrabella0 | 1
myrabella1 | 1
myrabella2 | 1
myrabella3 | 1
myrabella4 | 1
myrabella5 | 1
myrabella6 | 1
myrabella7 | 1
myrabella8 | 1
nasa0 | 1
nasa1 | 1
nasa2 | 1
nasa3 | 1
nasa4 | 1
nasa5 | 1
nasa6 | 1
nasa7 | 1
nasa8 | 1
nasa9 | 1
nasa10 | 1
nasa11 | 1
nasa12 | 1
nasa13 | 1
nasa28 | 1
nasa29 | 1
sea0 | 1
sea1 | 1
sea2 | 1
sea3 | 1
sea4 | 1
sea5 | 1
sea6 | 1
sea7 | 1
sea8 | 1
sea9 | 1
sea10-sfiera | 1
sea11-sfiera | 1
sea12-sfiera | 1
sea13-sfiera | 1
sea15-harro | 1
sea16-harro | 1
sea17-harro | 1
sea18 | 1
sea19 | 1
sivael0 | 1
sivael1 | 1
sivael2 | 1
sivael3 | 1
sivael4 | 1
sivael5 | 1
sivael6 | 1
sivael7 | 1
sivael8 | 1
sivael9 | 1
sky0 | 1
sky1 | 1
sky2 | 1
sky3 | 1
sky4 | 1
sky5 | 1
sky6 | 1
sky7 | 1
sky8 | 1
sky9 | 1
sky10 | 1
sky11 | 1
snow0 | 1
snow1 | 1
snow2 | 1
snow3 | 1
snow4 | 1
snow5 | 1
snow6 | 1
snow7 | 1
snow8 | 1
snow9 | 1
snow10 | 1
snow10-sfiera | 1
snow11-sfiera | 1
snow12 | 1
snow13 | 1
snow14 | 1
snow15 | 1
snow16 | 1
snow17 | 1
snow18 | 1
snow19 | 1
snow20 | 1
snow21 | 1
space1 | 1
space2 | 1
space3 | 1
space4 | 1
space6 | 1
space7 | 1
station0 | 1
station1 | 1
station2 | 1
station3 | 1
station4 | 1
station5 | 1
station6 | 1
station7 | 1
station8 | 1
station9 | 1
station10 | 1
station11 | 1
station12 | 1
station13 | 1
station14 | 1
station15 | 1
station16 | 1
station17 | 1
station18 | 1
station19 | 1
station20 | 1
station21 | 1
station22 | 1
station23 | 1
station24 | 1
station25 | 1
station26 | 1
station27 | 1
station28 | 1
station29 | 1
station30 | 1
station31 | 1
station32 | 1
station33 | 1
station34 | 1
station35 | 1
station36 | 1
station37 | 1
station38 | 1
station39 | 1
station40 | 1
station41 | 1
station42 | 1
station43 | 1
station44 | 1
station45 | 1
valley0 | 1
valley1 | 1
valley2 | 1
valley3 | 1
valley4 | 1
valley5 | 1
valley6 | 1
valley7 | 1
valley8 | 1
valley9 | 1
valley10-sfiera | 1
valley11-harro | 1
valley12-harro | 1
valley13-harro | 1
valley14-harro | 1
valley15-harro | 1
water0 | 1
water1 | 1
water2 | 1
water3 | 1
water4 | 1
water5 | 1
water6 | 1
water7 | 1
water8 | 1
water9 | 1
water10-harro | 1
water11-harro | 1
water12 | 1
water13 | 1
city15-sfiera | 0
city16-sfiera | 0
city17-sfiera | 0
clouds_00 | 0
clouds_01 | 0
clouds_02 | 0
clouds_03 | 0
clouds_04 | 0
clouds_05 | 0
clouds_06 | 0
clouds_07 | 0
clouds_08 | 0
desert9-sfiera | 0
enceladus_1 | 0
enceladus_2 | 0
enceladus_3 | 0
enceladus_4 | 0
mountain14-sfiera | 0
nasa14 | 0
nasa15 | 0
nasa16 | 0
nasa17 | 0
nasa18 | 0
nasa19 | 0
nasa20 | 0
nasa21 | 0
nasa22 | 0
nasa23 | 0
nasa24 | 0
nasa25 | 0
nasa26 | 0
nasa27 | 0
pia1 | 0
pia2 | 0
sea14-sfiera | 0
space5 | 0
</details>

### Alternative Approaches

N/A

### Additional Context

Some notes on certain landscape images:
* ~~space#, station#, and loc# get used a lot given the lack of fitting images that we have for stations.~~
* ~~station4~~ and space5 were at one point used but have since been replaced, likely as a result of these images containing people, which violates the landscape image guidelines.
* ~~space0, space2, and station3 were also used and replaced, but I'm not entirely sure why.~~
* The unused nasa#, pia#, enceladus#, and clouds# images are new and will likely be used by @Zitchas at some point.
* The unused sea14-sfiera image contains human architecture in the background. It may be possible to find a human world that already uses a sea# image that could use sea14. The images mountain14-sfiera and desert9-sfiera are in the same boat.
* The unused city#-sfiera images show human cities. It may be possible to find urban human worlds that don't use a city image that could use these images, putting the city images to use while also potentially freeing up a new image to be used elsewhere and reducing the number of duplicates. ~~The unused industrial0-iridium, which shows some sort of human industrial area with lots of trains, could be put to use under similar circumstances.~~
* ~~The unused snow20 is a spare from #6450.~~

**Comment by a-random-lemurian** [comment]
Would we want to modify the parser to throw a warning when it detects duplicate landscape images?

**Comment by Amazinite** [comment]
Updated the table after the changes from #6450. We now only have two images that are each used twice, with all other images being used either one or zero times.

**Comment by Amazinite** [comment]
Transferred to a discussion as opposed to having this be a perpetually open issue.

**Comment by roadrunner56** [comment]
Should we update this now that the Gegno and Bunrodea are in, seeing as those two created new duplicates? Or should we wait untill the Xapelaux are in too so we can do it all at once?

**Comment by Hecter94** [comment]
I ran a quick regex query to count how many times each landscape image is used, just to give us a general idea of where we're at. 
Here are the ones that are used multiple times:

<details>
<summary> Long Table </summary>

Landscape | Uses
-- | --
sky3 | 3
forest2 | 3
mountain5 | 3
badlands6 | 3
desert2 | 3
space2 | 3
canyon4 | 2
forest4 | 2
water6 | 2
canyon12 | 2
station33 | 2
mountain2 | 2
canyon5 | 2
sky6 | 2
sea2 | 2
canyon01 | 2
sivael6 | 2
fields0 | 2
snow13 | 2
station34 | 2
fields3 | 2
sea5 | 2
mountain1 | 2
desert4 | 2
badlands0 | 2
lava10 | 2
beach11-harro | 2
fog0 | 2
mountain4 | 2
badlands2 | 2
station36 | 2
canyon03 | 2
forest5 | 2
station2 | 2
forest3 | 2
sky5 | 2
station5 | 2
desert0 | 2
badlands8 | 2
station9 | 2
fields6 | 2
mountain0 | 2
fog6 | 2
industrial0-iridium | 2
fog2 | 2
canyon0 | 2
desert5 | 2
hills1 | 2
fog1 | 2
mars2 | 2
snow14 | 2
</details>

Also these are all apparently used 0 times, which is probably worth mentioning:
<details>
<summary>Table 2</summary>

Landscape | Uses
-- | --
beach7-sfiera | 0
city12 | 0
city15-sfiera | 0
city16-sfiera | 0
city17-sfiera | 0
city18-iridium | 0
city19 | 0
city3 | 0
clouds_00 | 0
clouds_01 | 0
clouds_02 | 0
clouds_04 | 0
clouds_05 | 0
clouds_06 | 0
clouds_07 | 0
clouds_08 | 0
desert9-sfiera | 0
enceladus_1 | 0
enceladus_2 | 0
enceladus_3 | 0
enceladus_4 | 0
fields16 | 0
fields17 | 0
fields18 | 0
fields19 | 0
fields20 | 0
fields21 | 0
fields22 | 0
fields23 | 0
fields24 | 0
fields25 | 0
fields26 | 0
fields27 | 0
hills10 | 0
mountain14-sfiera | 0
nasa14 | 0
nasa15 | 0
nasa16 | 0
nasa17 | 0
nasa18 | 0
nasa19 | 0
nasa20 | 0
nasa21 | 0
nasa22 | 0
nasa23 | 0
nasa24 | 0
nasa25 | 0
nasa26 | 0
nasa27 | 0
pia21553_1 | 0
pia21553_2 | 0
sea14-sfiera | 0
sea20 | 0
sea21 | 0
space0 | 0
space5 | 0
valley6 | 0
water15 | 0
water16 | 0
water17 | 0
water18 | 0
water19 | 0
water20 | 0

</details>

**Comment by warp-core** [comment]
<details> <summary>A better table</summary>

sprite | planets using it
-- | --
land/badlands0 | Dustmaker,Illbo Elo,
land/badlands2 | Esstch,Tik Klai,
land/badlands6 | Devil-Eye,Grakhord,Illbo Avo,
land/badlands8 | Melenci,Mutiny,
land/beach11-harro | Ef Osch,Weir of Glubatub,
land/canyon0 | Revelation,Yniu Eiu,
land/canyon01 | Ceaseless Toil,Enn Bue,
land/canyon03 | Forpelog,Yniu Ena,
land/canyon12 | Arroharg,Babiali,
land/canyon4 | Ada,Oup Je,
land/canyon5 | Bosunothro,Varu K'est,
land/desert0 | Mars,Var Oti,
land/desert2 | Echo,Iemn Eitch,Kisarra,
land/desert4 | Dueyu Eitch,Farpoint,
land/desert5 | Serpens,Vigales,
land/fields0 | Cornucopia,Deli Kat,
land/fields3 | Deli Kasi,Lichen,
land/fields6 | New Kansas,Sardva,
land/fog0 | Enbye,Heartvalley,
land/fog1 | Ternituul,Warm Wind,
land/fog2 | Okoity,Seleptra Nak,
land/fog6 | New Tortuga,Nimbus,
land/forest2 | Blubipad's Workshop,Disara,Graede,
land/forest3 | Gue Faur,Pearl,
land/forest4 | Ahr,Kalbai,
land/forest5 | Gentle Rain,Siedi,
land/hai | Hai,Hai Wanderer,
land/hills1 | Sestor Ikfar,Sies Upi,
land/industrial0-iridium | Oblivion,Sopoyra,
land/lava10 | Ef Aourtdye,Vatican,
land/loc1 | Scar's Hideout,Smuggler's Den,
land/mars2 | Tibernia,Truklar,
land/mountain0 | New Tibet,Thshybothro,
land/mountain1 | Dueitch Ae,Shroud,
land/mountain2 | Bluestone,Dlia Jzaur,
land/mountain4 | Essime,Harmony,
land/mountain5 | Cipi,Erabuthro,New Switzerland,
land/sea17-harro | Aspiration,Avalon,
land/sea2 | Byruca,Laki Nemparu,
land/sea5 | Drekag Firask,Eragarthro,
land/sivael6 | Charon Station,Tschyss,
land/sky3 | Ablub's Invention,Gara Pvu,Olohoa,
land/sky5 | Hestia,Jentuthro,
land/sky6 | Bunthro,Featherweight,
land/snow13 | Cyife,Seraglio,
land/snow14 | Var' Roi,Wye Tzeou,
land/space2 | Elenchus,Iddesato,Station Cian,
land/station2 | Giaru Gegno,Ring of Wisdom,
land/station33 | Bluerun,Guardian Array Sapphire,
land/station34 | Darkrest,Periaxle Circuit,
land/station36 | Firefall,Ogmur's Siphon,
land/station5 | Lagrange,Ruelogakk,
land/station9 | Muninn Station,Silo of Ablodab,
land/water6 | Angko,Darkcloak,


</summary>

**Comment by julzaa** [comment]
As of now:

| Image               | Usage |
| ------------------- | ----- |
| station25           | 4     |
| sea2                | 4     |
| mars2               | 4     |
| sky3                | 3     |
| mountain26          | 3     |
| station16           | 3     |
| mountain5           | 3     |
| badlands0           | 3     |
| desert2             | 3     |
| canyon03            | 3     |
| forest5             | 3     |
| space2              | 3     |
| desert13            | 3     |
| fields6             | 3     |
| canyon15            | 2     |
| canyon4             | 2     |
| nasa30              | 2     |
| forest4             | 2     |
| station48           | 2     |
| space7              | 2     |
| canyon13            | 2     |
| station49           | 2     |
| water6              | 2     |
| mars3               | 2     |
| snow20              | 2     |
| canyon3             | 2     |
| canyon12            | 2     |
| haze1               | 2     |
| station33           | 2     |
| mountain2           | 2     |
| canyon5             | 2     |
| sky7                | 2     |
| canyon1             | 2     |
| lava8               | 2     |
| canyon01            | 2     |
| sivael6             | 2     |
| snow7               | 2     |
| desert3             | 2     |
| myrabella0          | 2     |
| fields0             | 2     |
| nasa7               | 2     |
| station34           | 2     |
| sea10-sfiera        | 2     |
| fields3             | 2     |
| hills0              | 2     |
| station27           | 2     |
| forest2             | 2     |
| sea5                | 2     |
| mountain1           | 2     |
| desert4             | 2     |
| mountain9           | 2     |
| beach11-harro       | 2     |
| mountain4           | 2     |
| badlands2           | 2     |
| station36           | 2     |
| dune0               | 2     |
| clouds_03           | 2     |
| station2            | 2     |
| badlands6           | 2     |
| water3              | 2     |
| forest3             | 2     |
| lava1               | 2     |
| sky5                | 2     |
| myrabella8          | 2     |
| station42           | 2     |
| nasa2               | 2     |
| mars1               | 2     |
| mars0               | 2     |
| station5            | 2     |
| loc3                | 2     |
| sky1                | 2     |
| desert0             | 2     |
| myrabella4          | 2     |
| fields14            | 2     |
| station9            | 2     |
| badlands4           | 2     |
| mountain0           | 2     |
| hills4              | 2     |
| industrial0-iridium | 2     |
| fog2                | 2     |
| dmottl1             | 2     |
| station30           | 2     |
| desert12            | 2     |
| moon0               | 2     |
| bwerner6            | 2     |
| beach3              | 2     |
| canyon0             | 2     |
| desert15            | 2     |
| lava3               | 2     |
| fog1                | 2     |
| snow14              | 2     |
| water4              | 2     |
